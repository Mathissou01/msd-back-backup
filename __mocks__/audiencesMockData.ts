import { GetAudiencesByContractIdDocument } from "../src/graphql/codegen/generated-types";

export const getNoAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [],
        },
      },
    },
  },
];

export const getSingleAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getTwoAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getAllAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
            {
              id: 3,
              attributes: {
                isActive: true,
                type: "Professionnels",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];
