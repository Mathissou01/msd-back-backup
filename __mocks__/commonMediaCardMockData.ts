import { GetFilesPaginationByFolderIdDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetFilesPaginationByFolderIdDocument,
      filters: {
        folder: {
          id: {
            eq: 2,
          },
        },
      },
      pagination: {
        page: 1,
        pageSize: 10,
      },
      sort: "createdAt:desc",
    },
    result: {
      data: {
        uploadFiles: {
          data: [
            {
              id: "124",
              attributes: {
                name: "La Bruja.jpeg",
                mime: "image/jpeg",
                size: 124.35,
                width: 1280,
                height: 853,
                createdAt: "2023-01-26T10:34:26.058Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/La%20Bruja.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "122",
              attributes: {
                name: "Banana Mama Rooftop & Kitchen 5.jpeg",
                mime: "image/jpeg",
                size: 68.84,
                width: 960,
                height: 960,
                createdAt: "2023-01-26T10:30:22.701Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Banana%20Mama%20Rooftop%20%26%20Kitchen%205.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "121",
              attributes: {
                name: "Sichuan Hot Pot.jpeg",
                mime: "image/jpeg",
                size: 2233.13,
                width: 6962,
                height: 4615,
                createdAt: "2023-01-18T17:39:30.518Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Sichuan%20Hot%20Pot.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "120",
              attributes: {
                name: "mahbod-akhzami-kg40JrEsCvY-unsplash.jpg",
                mime: "image/jpeg",
                size: 1788.48,
                width: 6000,
                height: 4000,
                createdAt: "2023-01-18T17:39:29.900Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/mahbod-akhzami-kg40JrEsCvY-unsplash.jpg",
                ext: ".jpg",
              },
            },
            {
              id: "119",
              attributes: {
                name: "Paella in Valencia.jpeg",
                mime: "image/jpeg",
                size: 1026.93,
                width: 3851,
                height: 2528,
                createdAt: "2023-01-18T17:39:29.720Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Paella%20in%20Valencia.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "118",
              attributes: {
                name: "Rusconi's American Kitchen 3.jpeg",
                mime: "image/jpeg",
                size: 209.81,
                width: 1999,
                height: 1333,
                createdAt: "2023-01-18T17:39:29.502Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Rusconi%27s%20American%20Kitchen%203.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "117",
              attributes: {
                name: "Quince 5.jpeg",
                mime: "image/jpeg",
                size: 197.32,
                width: 2000,
                height: 1500,
                createdAt: "2023-01-18T17:39:29.310Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Quince%205.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "116",
              attributes: {
                name: "Rusconi's American Kitchen 2.jpeg",
                mime: "image/jpeg",
                size: 72.49,
                width: 1000,
                height: 1500,
                createdAt: "2023-01-18T17:39:29.221Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Rusconi%27s%20American%20Kitchen%202.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "115",
              attributes: {
                name: "Pur' - Jean-Fran√ßois Rouquette 3.jpeg",
                mime: "image/jpeg",
                size: 63.2,
                width: 1280,
                height: 853,
                createdAt: "2023-01-18T17:39:29.039Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/Pur%27%20-%20Jean-Fran%C3%A7ois%20Rouquette%203.jpeg",
                ext: ".jpeg",
              },
            },
            {
              id: "114",
              attributes: {
                name: "muneeb-muhamed-kPXT9HA4LOk-unsplash.jpg",
                mime: "image/jpeg",
                size: 516.64,
                width: 3456,
                height: 4608,
                createdAt: "2023-01-18T17:39:28.922Z",
                url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-sdial/muneeb-muhamed-kPXT9HA4LOk-unsplash.jpg",
                ext: ".jpg",
              },
            },
          ],
          meta: {
            pagination: {
              total: 122,
              pageSize: 10,
              pageCount: 13,
              page: 1,
            },
          },
        },
      },
    },
  },
];
