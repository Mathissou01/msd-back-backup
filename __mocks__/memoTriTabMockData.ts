import { GetRecyclingGuideServiceByIdDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetRecyclingGuideServiceByIdDocument,
      variables: {
        recyclingGuideServiceId: "1",
      },
    },
    result: {
      data: {
        recyclingGuideService: {
          data: {
            id: "1",
            attributes: {
              name: "Guide du tri",
              orderExtension: true,
              memoName: "Nom du MÃ©motri test",
              memoDesc: "desc",
              memoFile: {
                data: {
                  id: "197",
                  attributes: {
                    hash: "sample_4d68bee0d1",
                    mime: "application/pdf",
                    name: "sample.pdf",
                    provider: "azure-storage-blob",
                    size: 3.04,
                    url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-hamna/sample.pdf",
                    alternativeText: "sample.pdf",
                    ext: ".pdf",
                    height: null,
                    width: null,
                    createdAt: "2023-05-23T16:20:29.293Z",
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
