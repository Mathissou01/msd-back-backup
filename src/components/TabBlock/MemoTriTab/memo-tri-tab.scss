@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.shadows" as shadows;

.c-MemoTriTab {
  position: relative;
  width: 100%;
  height: 100%;

  &__Title {
    margin-bottom: 18px;
    font-size: 28px;
  }

  &__Form {
    display: flex;
    flex-direction: column;
    gap: 24px;
    width: 100%;
  }

  &__Group {
    display: flex;
    flex-direction: column;
    gap: 10px;

    &_short {
      width: 60%;
    }
  }

  &__Buttons {
    display: flex;
    gap: 16px;
  }
}
