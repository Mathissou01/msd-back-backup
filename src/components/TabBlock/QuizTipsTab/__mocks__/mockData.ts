import { GetQuizAndTipsBlockDocument } from "../../../../graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      quizAndTipsBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            title: "Titre modifi√©",
                            displayBlock: false,
                            displayQuiz: false,
                            quiz: {
                              data: {
                                id: "2",
                                attributes: {
                                  title: "Nom du quiz",
                                  publishedAt: "2022-11-20T02:34:14.535Z",
                                },
                              },
                            },
                            displayTips: false,
                            tips: {
                              data: [],
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
        services: {
          data: [
            {
              attributes: {
                serviceInstance: [
                  {
                    __typename: "ComponentMsdEditorial",
                    id: "4",
                    editorialServices: {
                      data: [
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoQuizzesSubService",
                                isActivated: true,
                                quizzes: {
                                  data: [
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Nom du quiz",
                                        publishedAt: "2022-11-20T02:34:14.535Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "truc",
                                        publishedAt: "2022-11-18T10:16:08.936Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoTipsSubService",
                                isActivated: true,
                                tips: {
                                  data: [
                                    {
                                      id: "5",
                                      attributes: {
                                        title: "nah",
                                        publishedAt: "2022-11-19T00:14:37.403Z",
                                      },
                                    },
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Titre de l'astuce 1",
                                        publishedAt: "2022-11-19T00:14:08.232Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "Titre de l'astuce 2",
                                        publishedAt: "2022-11-18T10:17:11.158Z",
                                      },
                                    },
                                    {
                                      id: "3",
                                      attributes: {
                                        title: "Titre de l'astuce 3",
                                        publishedAt: "2022-11-19T00:14:20.490Z",
                                      },
                                    },
                                    {
                                      id: "4",
                                      attributes: {
                                        title: "Titre de l'astuce 4",
                                        publishedAt: "2022-11-19T00:14:30.273Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    error: new Error("An error occured"),
  },
];
