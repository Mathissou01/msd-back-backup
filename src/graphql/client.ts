import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache({
      typePolicies: {
        UploadFolder: {
          merge: true,
        },
        UploadFile: {
          merge: true,
        },
        Contract: {
          merge: true,
        },
        ContractCustomization: {
          merge: true,
        },
        New: {
          merge: true,
        },
        Tip: {
          merge: true,
        },
        RecyclingGuideService: {
          merge: true,
        },
        WasteForm: {
          merge: true,
        },
        WasteFamily: {
          merge: true,
        },
        Flow: {
          merge: true,
        },
        FlowColor: {
          merge: true,
        },
        DropOffMap: {
          merge: true,
        },
        Request: {
          merge: true,
        },
        RequestAggregate: {
          merge: true,
        },
      },
    }),
  });
}

export default client;
