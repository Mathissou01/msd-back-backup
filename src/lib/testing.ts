import { prettyDOM } from "@testing-library/react";

export function removeUnstableHtmlPropertiesForTesting(
  htmlElement: HTMLElement,
  attributeName: string,
) {
  const domHTML = prettyDOM(htmlElement, Infinity).toString();
  if (!domHTML) return undefined;
  const regex = new RegExp(`${attributeName}(.*)"(.*)"`, "g");
  return domHTML.replaceAll(regex, "");
}
