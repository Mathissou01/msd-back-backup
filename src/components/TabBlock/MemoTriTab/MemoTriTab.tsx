import React, { useEffect } from "react";
import { FieldValues, FormProvider, useForm } from "react-hook-form";
import {
  fileSizeLimitationOptions,
  remapUploadFileEntityToLocalFile,
} from "../../../lib/media";
import { useContract } from "../../../hooks/useContract";
import { useFocusFirstElement } from "../../../hooks/useFocusFirstElement";
import {
  GetRecyclingGuideServicesByContractIdDocument,
  useGetRecyclingGuideServiceByIdLazyQuery,
  useUpdateRecyclingGuideServiceByIdMutation,
} from "../../../graphql/codegen/generated-types";
import FormInput from "../../Form/FormInput/FormInput";
import FormFileInput from "../../Form/FormFileInput/FormFileInput";
import CommonButton from "../../Common/CommonButton/CommonButton";
import CommonLoader from "../../Common/CommonLoader/CommonLoader";
import "./memo-tri-tab.scss";

export default function MemoTriTab() {
  /* Static Data */
  const formLabels = {
    title: "Mémotri",
    maxCharactersLabel: "caractères maximum",
    memoName: "Nom du mémotri",
    memoDesc: "Description du mémotri",
    memoFile: "Fichier",
    staticImageValidation: "20 Mo maximum",
    staticImagePlaceholder:
      "Cliquer pour ajouter un fichier depuis la bibliothèque de média ou glissez-déposez un fichier dans cette zone.",
    submitButtonLabel: "Enregistrer les modifications",
    cancelButtonLabel: "Annuler les modifications",
  };
  const memoNameMaxChar = 20;
  const memoDescMaxChar = 50;

  /* Local Data */
  const form = useForm<FieldValues>({
    mode: "onChange",
  });
  const {
    handleSubmit,
    formState: { isDirty, isSubmitting, isValid },
  } = form;

  /* External Data */
  const { contractId, contract } = useContract();
  const [getGetRecyclingGuideServiceById, { data, loading, error }] =
    useGetRecyclingGuideServiceByIdLazyQuery();
  const [
    updateRecyclingGuideService,
    { loading: mutationLoading, error: mutationError },
  ] = useUpdateRecyclingGuideServiceByIdMutation();

  /* Methods */
  async function onSubmitValid(submitData: FieldValues) {
    if (submitData.id) {
      const variables = {
        updateRecyclingGuideServiceId: submitData.id,
        data: {
          memoName: submitData.memoName,
          memoDesc: submitData.memoDesc,
          memoFile: submitData.memoFile.id,
        },
      };
      return updateRecyclingGuideService({
        variables,
        refetchQueries: [
          {
            query: GetRecyclingGuideServicesByContractIdDocument,
            variables: { contractId },
          },
        ],
      });
    }
  }

  function onCancel() {
    form.reset();
  }

  useEffect(() => {
    if (contract.attributes?.recyclingGuideService?.data?.id) {
      getGetRecyclingGuideServiceById({
        variables: {
          recyclingGuideServiceId:
            contract.attributes.recyclingGuideService.data.id,
        },
      });
    }
  }, [contract, getGetRecyclingGuideServiceById]);

  useEffect(() => {
    if (data && data.recyclingGuideService?.data) {
      const recyclingGuideService = data.recyclingGuideService.data;
      if (recyclingGuideService.id && recyclingGuideService.attributes) {
        const mappedData = {
          id: recyclingGuideService.id,
          memoName: recyclingGuideService.attributes.memoName,
          memoDesc: recyclingGuideService.attributes.memoDesc,
          memoFile: remapUploadFileEntityToLocalFile(
            recyclingGuideService.attributes.memoFile?.data,
          ),
        };
        form.reset(mappedData);
      }
    }
  }, [form, data]);

  return (
    <div
      className="c-MemoTriTab"
      id="panel-memoTri"
      role="tabpanel"
      aria-labelledby="tab-memoTri"
    >
      <CommonLoader
        isLoading={loading || isSubmitting || mutationLoading}
        isShowingContent={isSubmitting || mutationLoading}
        hasDelay={isSubmitting || mutationLoading}
        errors={[error, mutationError]}
      >
        <h2 className="c-MemoTriTab__Title">{formLabels.title}</h2>
        <FormProvider {...form}>
          <form
            className="c-MemoTriTab__Form"
            onSubmit={handleSubmit(onSubmitValid)}
            ref={useFocusFirstElement()}
          >
            <div className="c-MemoTriTab__Group c-MemoTriTab__Group_short">
              <FormInput
                type="text"
                name="memoName"
                label={formLabels.memoName}
                validationLabel={`${memoNameMaxChar} ${formLabels.maxCharactersLabel}`}
                maxLengthValidation={memoNameMaxChar}
                isRequired
              />
            </div>
            <div className="c-MemoTriTab__Group c-MemoTriTab__Group_short">
              <FormInput
                type="text"
                name="memoDesc"
                label={formLabels.memoDesc}
                validationLabel={`${memoDescMaxChar} ${formLabels.maxCharactersLabel}`}
                maxLengthValidation={memoDescMaxChar}
                tagType="textarea"
              />
            </div>
            <div className="c-MemoTriTab__Group c-MemoTriTab__Group_short">
              <FormFileInput
                name="memoFile"
                label={formLabels.memoFile}
                isRequired
                validationLabel={formLabels.staticImageValidation}
                fileSizeLimitation={fileSizeLimitationOptions._20mb}
                placeholder={formLabels.staticImagePlaceholder}
              />
            </div>

            <div className="c-MemoTriTab__Buttons">
              <CommonButton
                type="submit"
                label={formLabels.submitButtonLabel}
                style="primary"
                picto="check"
                isDisabled={!isValid}
              />
              <CommonButton
                type="button"
                label={formLabels.cancelButtonLabel}
                picto="cross"
                onClick={onCancel}
                isDisabled={!isDirty}
              />
            </div>
          </form>
        </FormProvider>
      </CommonLoader>
    </div>
  );
}
