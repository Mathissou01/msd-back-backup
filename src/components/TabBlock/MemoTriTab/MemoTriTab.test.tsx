import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import React from "react";
import { defaultMockData } from "../../../../__mocks__/memoTriTabMockData";
import MemoTriTab from "./MemoTriTab";

jest.mock("../../../graphql/client", () => null);
describe("MemoTriTab", () => {
  it("renders loading and loaded state", async () => {
    const { container } = render(
      <MockedProvider mocks={defaultMockData}>
        <MemoTriTab />
      </MockedProvider>,
    );

    expect(await container).toMatchSnapshot();
    expect(await screen.findByText("Nom du mémotri *")).toBeInTheDocument();
    expect(
      await screen.findByText("20 caractères maximum"),
    ).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
  });
});
