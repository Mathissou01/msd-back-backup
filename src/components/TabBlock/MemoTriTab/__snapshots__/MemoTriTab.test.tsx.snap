// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoTriTab renders loading and loaded state 1`] = `
<div>
  <div
    aria-labelledby="tab-memoTri"
    class="c-MemoTriTab"
    id="panel-memoTri"
    role="tabpanel"
  >
    <div
      class="c-CommonLoader c-CommonLoader_cover c-CommonLoader_grow"
    >
      <div
        class="c-CommonSpinner"
        data-testid="common-spinner"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`MemoTriTab renders loading and loaded state 2`] = `
<div>
  <div
    aria-labelledby="tab-memoTri"
    class="c-MemoTriTab"
    id="panel-memoTri"
    role="tabpanel"
  >
    <h2
      class="c-MemoTriTab__Title"
    >
      Mémotri
    </h2>
    <form
      class="c-MemoTriTab__Form"
    >
      <div
        class="c-MemoTriTab__Group c-MemoTriTab__Group_short"
      >
        <div
          class="c-FormInput"
        >
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="memoName"
            >
              <span
                class="c-FormLabel__Label"
              >
                Nom du mémotri *
              </span>
              <span
                class="c-FormLabel__Validation"
              >
                20 caractères maximum
              </span>
            </label>
            <div
              class="c-FormLabel__Content"
            >
              <input
                aria-errormessage="memoName_error"
                aria-invalid="false"
                class="c-FormInput__Input"
                data-testid="form-input"
                id="memoName"
                maxlength="20"
                name="memoName"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-MemoTriTab__Group c-MemoTriTab__Group_short"
      >
        <div
          class="c-FormInput"
        >
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="memoDesc"
            >
              <span
                class="c-FormLabel__Label"
              >
                Description du mémotri
              </span>
              <span
                class="c-FormLabel__Validation"
              >
                50 caractères maximum
              </span>
            </label>
            <div
              class="c-FormLabel__Content"
            >
              <textarea
                aria-errormessage="memoDesc_error"
                aria-invalid="false"
                class="c-FormInput__Input"
                data-testid="form-input"
                id="memoDesc"
                maxlength="50"
                name="memoDesc"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-MemoTriTab__Group c-MemoTriTab__Group_short"
      >
        <div
          class="c-FormFileInput"
        >
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="memoFile"
            >
              <span
                class="c-FormLabel__Label"
              >
                Fichier *
              </span>
              <span
                class="c-FormLabel__Validation"
              >
                20 Mo maximum
              </span>
            </label>
            <div
              class="c-FormLabel__Content"
            >
              <input
                aria-invalid="false"
                name="memoFile"
                type="hidden"
                value=""
              />
              <div
                aria-errormessage="memoFile_error"
                aria-invalid="false"
                class="c-FormFileInput__DragDrop"
                id="memoFile"
              >
                <span
                  class="c-FormFileInput__Placeholder"
                >
                  Cliquer pour ajouter un fichier depuis la bibliothèque de média ou glissez-déposez un fichier dans cette zone.
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-MemoTriTab__Buttons"
      >
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_primary c-CommonButton_fontSmall"
          disabled=""
          type="submit"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_check"
          />
          Enregistrer les modifications
        </button>
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_secondary c-CommonButton_fontSmall"
          disabled=""
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_cross"
          />
          Annuler les modifications
        </button>
      </div>
    </form>
  </div>
</div>
`;
